pkg/plugin/processor/builtin/renameField.go

```diff
@@ -0,0 +1,149 @@
+		Author:  "Meroxa, Inc.",
+		Parameters: map[string]sdk.Parameter{
+			"mapping": {
+				Default: "",
+				Type:    sdk.ParameterTypeString,
+				Description: "a comma separated list of keys and values, for fields and their new names, " +
+					"ex: .Metadata.key:id,.Payload.After.foo:bar",
+				Validations: []sdk.Validation{
+					{
+						Type: sdk.ValidationTypeRequired,
+					}, {
+						Type: sdk.ValidationTypeExclusion,
+						Value: MetadataReference + "," + PayloadReference + "," + PayloadBeforeReference + "," +
+							PayloadAfterReference + "," + PositionReference + KeyReference + "," + OperationReference,
+					},
+				},
+			},
+		},
+	}, nil
+}
+
+func (p *renameField) Configure(_ context.Context, cfg map[string]string) error {
+	list, ok := cfg["mapping"]
+	if !ok || list == "" {
+		return cerrors.Errorf("%w (%q)", ErrRequiredParamMissing, "mapping")
+	}
+
+	result := make(map[string]string)
+	pairs := strings.Split(list, ",")
+	for _, pair := range pairs {
+		parts := strings.Split(pair, ":")
+		if len(parts) != 2 {
+			return cerrors.Errorf("wrong format for the %q param, should be a comma separated list of keys and values,"+
+				"ex: .Metadata.key:id,.Payload.After.foo:bar", "mapping")
+		}
+		key := strings.TrimSpace(parts[0])
+		if key == OperationReference || key == KeyReference || key == PayloadReference || key ==
+			PayloadBeforeReference || key == PayloadAfterReference || key == PositionReference || key == MetadataReference {
+			return cerrors.Errorf("cannot rename one of the top-level fields %q", key)
+		}
+		value := strings.TrimSpace(parts[1])
+		result[key] = value
+	}
+	p.mapping = result
+
+	return nil
+}
+
+func (p *renameField) Open(context.Context) error {
+	return nil
+}
+
+func (p *renameField) Process(_ context.Context, records []opencdc.Record) []sdk.ProcessedRecord {
+	out := make([]sdk.ProcessedRecord, 0, len(records))
+	for _, record := range records {
+		for key, val := range p.mapping {
+			err := p.rename(record, key, val)
+			if err != nil {
+				return append(out, sdk.ErrorRecord{Error: err})
+			}
+		}
+		out = append(out, sdk.SingleRecord(record))
+	}
+	return out
+}
+
+func (p *renameField) rename(record opencdc.Record, key, val string) error {
+	resolver1, err := sdk.NewReferenceResolver(key)
+	if err != nil {
+		return err
+	}
+	ref1, err := resolver1.Resolve(&record)
+	if err != nil {
+		return err
+	}
+	resolver2, err := sdk.NewReferenceResolver(p.getNameWithPrefix(val, key))
+	if err != nil {
+		return err
+	}
+	// create a second reference to the new name
+	ref2, err := resolver2.Resolve(&record)
+	if err != nil {
+		return err
+	}
+	// copy the value over to the new name
+	err = ref2.Set(ref1.Get())
+	if err != nil {
+		return err
+	}
+	// delete the old name field
+	err = ref1.Delete()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (p *renameField) getNameWithPrefix(newName, oldName string) string {
+	// split the oldName by dots
+	parts := strings.Split(oldName, ".")
```

Comment:
What if you have something like `.Metadata["old name"]`?

Reply:
oof good catch! didn't even occur to me, will handle that too

Reply:
Good point! We might need to expand the resolver with a method that returns a resolver with a renamed field, it would be trivial to do there.